crypto/asn1/a_gentm.c

```diff
@@ -271,3 +322,38 @@ int ASN1_GENERALIZEDTIME_print(BIO *bp, const ASN1_GENERALIZEDTIME *tm)
     BIO_write(bp, "Bad time value", 14);
     return (0);
 }
+
+int ASN1_GENERALIZEDTIME_print_gmt(BIO *bp, const ASN1_GENERALIZEDTIME *t)
```

Comment:
The existing function ASN1_GENERALIZEDTIME_print() does not properly deal with GMT (it's listed as a bug in the documentation) as it discards any offset (which is legal outside a certificate) and basically prints it as local time. It's actually quite useless. This forces a ASN1_GENERALIZEDTIME to always be printed as GMT.